tensorlayer/iterate.py

```diff
@@ -59,7 +59,10 @@ def minibatches(inputs=None, targets=None, batch_size=None, shuffle=False):
             excerpt = indices[start_idx:start_idx + batch_size]
         else:
             excerpt = slice(start_idx, start_idx + batch_size)
-        yield inputs[excerpt], targets[excerpt]
+        if (isinstance(inputs, list) or isinstance(targets, list)) and (shuffle == True):
```

Comment:
does it still work if `isinstance(inputs, list) == True` but `isinstance(targets, list) == False`?

Reply:
nope, it doesn't, list can't be indexed by a list of ID.

Reply:
`inputs = [inputs[i] for i in excerpt] if isinstance(inputs, list) else inputs[excerpt]
targets = [targets[i] for i in excerpt] if isinstance(targets, list)) else targets[excerpt]
yield inputs, targets `